### ISSUE TITLE
Improvement:  Make WebMock ignore the last slash (/) in URLs

### ISSUE THREAD

**Issue by KelseyDH** [bblimke/webmock#864]
A common mistake when testing API's is accidentally including a `/` in a url in one place, while omitting that `/` in another endpoint.  Effectively the endpoint works, yet WebMock will say they are different. 

 E.g. writing `www.example.com/foo` in `assert_requested` when the actual API hits `http://example.com/foo/`. When POSTing to an endpoint with parameters, this slight difference in HTTP configurations is easy to miss and can lead to many hours of frustration.

---

My suggestion is that WebMock treats URL's ending in `/` the same as those omitting it.  For those wanting the old behaviour, the gem could add a flag to reenable it for the few odd projects actually treating calls to `http://example.com/foo/` and `http://example.com/foo' differently.

**Comment by fcheung** [comment]
Isn’t the flip side to this that you could have a passing test but code the fails in production because of extra or missing / that would be equally hard to spot?

**Comment by KelseyDH** [comment]
@fcheung  That would happen if the server treats `http://example.com/foo/` & `http://example.com/foo` differently.  My sense with a majority of Rails apps is that this is not the case.  Usually though -- you would have tested the actual code making the api call against the test server already, so the WebMock is just covering you from needlessly making the call repeatedly.

If the fear of this tripping up people is a problem though, perhaps this option could be enabled as a configuration option set during initialization?

**Comment by bblimke** [comment]
> My sense with a majority of Rails apps is that this is not the case

I agree though please take into account that often Rails apps are only clients of externals API's, non-Rails API's and WebMock is to verify the usage of these API's.

I'm not really sure how many API's treat trailing slash or lack of it as the same endpoint. I acknowledge bot your  @KelseyDH and @fcheung points. Perhaps WebMock should suggest adding or removing the missing slash in case it spots that this is the only difference?
